aiogram/types/message.py

```diff
@@ -1714,6 +1714,26 @@ def copy_to(
             reply_markup=reply_markup,
         )
 
+    def get_url(self, force_private: bool = False) -> str:
+        """
+        Returns message URL. Cannot be used in private (one-to-one) chats.
+        If chat has a username, returns URL like https://t.me/username/message_id
+        Otherwise (or if {force_private} flag is set), returns https://t.me/c/shifted_chat_id/message_id
+
+        :param force_private: if set, a private URL is returned even for a public chat
+        :return: string with full message URL
+        """
+        if self.chat.type == "private":
+            raise TypeError("Invalid chat type!")
+
+        url = "https://t.me/"
+        if not self.chat.username or force_private:
+            url += f"c/{self.chat.shifted_id}/"
```

Comment:
f-string + concatenation = bad solution
Since you are using an f-string anyway, just add previous text inside

Reply:
I can replace this with a single template for `.format()`

Reply:
something like `https://t.me/{chat_value}/{message_id}`
Where chat_value is either `/c/` + shifted ID or a username

Reply:
Why `.format()`? 
f-string is better solution. 
Define `chat_value` first and then use f-string